lib/order.rb

```diff
@@ -1,18 +1,57 @@
+require 'csv'
+require 'awesome_print'
+
 module Grocery
   class Order
     attr_reader :id, :products
-
-    def initialize(id, products)
+    def initialize(id, products) #products has to be a hash
       @id = id
       @products = products
     end
 
     def total
-      # TODO: implement total
+      @sum = 0
+      @products.each_value do |value|
+        @sum += value
+      end
+      @total = @sum + (@sum * 0.075).round(2)
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      if @products.key?(product_name)
+        return false
+      else
+        @products[product_name] = product_price
+        return true
+      end
+    end
+
+    def remove_product(product_name)
+      unless @products.key?(product_name)
+        return false
+      end
+      @products.delete(product_name)
+      return true
+    end
+
+    def self.all
+      all_orders = []
+      CSV.open("./support/orders.csv", 'r').each do |line|
+        products_hash = {}
+        id = line.slice!(0).to_i
+        line.join.split(";").each do |product| #Eliminate "outer" array. Split resulting string into array of products.
+          products_hash[product.split(":")[0]] = product.split(":")[1].to_f #create key, value pair for name, price of product
+        end
+        all_orders << Grocery::Order.new(id, products_hash)
+      end
+      return all_orders
+    end
+
+    def self.find(id)
+      if id.to_i < 1 || id.to_i > 100
```

Comment:
As mentioned above, you might want to not hardcode this lower and upper limit, in case you want to reuse this method for a different data set